Assets/00_Content/Scripts/Rounds/ScoreCard.cs

```diff
@@ -18,17 +25,53 @@ public class ScoreCard : MonoBehaviour {
 			set { roundNumber = Mathf.RoundToInt(roundNumber); }
 		}
 
-		private void OnEnable() {
-			if (EventSystem.current != null)
-				EventSystem.current.SetSelectedGameObject(startButton.gameObject);
+		private void Awake() {
+			for (int i = 0; i < readyCards.Length; i++) {
+				readyCards[i] = Instantiate(readyCardPrefab, readyCardRoot);
+				readyCards[i].Name = "Player " + i;
+				readyCards[i].gameObject.SetActive(false);
+			}
+		}
+
+		private void OnDisable() {
+			for (int i = 0; i < playerCount; i++) {
+				playerHandlers[i].player.OnStart.RemoveListener(playerHandlers[i].handler);
+				playerHandlers[i] = default;
+			}
 		}
 
 		public void UpdateScoreCard(int round) {
 			roundNumber = round;
 			roundNumberText.text = RoundNumber.ToString();
 		}
 
-		public void GoToNextRound() {
+		public void Show() {
+			playerCount = PlayerManager.Instance.Players.Count;
+			for (int i = 0; i < readyCards.Length; i++) {
+				readyCards[i].gameObject.SetActive(i < playerCount);
+				readyCards[i].Ready = false;
+			}
```

Comment:
Just curious, couldn't this be combined with the loop below?

Reply:
This loop is 0..MaxPlayers (see readyCards declaration), the other is 0..PlayerCount.

Reply:
Yes but is 0..Maxplayers needed? Is it for deactivating cards if a player suddenly disconnects or another reason?

Reply:
Hmm, yes. I suppose that was what I thought. But I guess that will probably never happen.